src/com/todddavies/components/progressbar/ProgressWheel.java

```diff
@@ -117,11 +117,18 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
         // and when we set the dimension we add it back again. Now the actual content 
         // of the view will be square, but, depending on the padding, the total dimensions 
         // of the view might not be.
-        if (widthWithoutPadding > heigthWithoutPadding) {
-            size = heigthWithoutPadding;
+        int heightMode = MeasureSpec.getMode(heightMeasureSpec);
+        int widthMode = MeasureSpec.getMode(widthMeasureSpec);
+        if (heightMode != MeasureSpec.UNSPECIFIED && widthMode != MeasureSpec.UNSPECIFIED) {
+            if (widthWithoutPadding > heigthWithoutPadding) {
+                size = heigthWithoutPadding;
+            } else {
+                size = widthWithoutPadding;
+            }

```

Comment:
Could this be:
    
    size = Math.min(heigthWithoutPadding, widthWithoutPadding);